pkg/processor/sourceprocessor/source_category_filler.go

```diff
@@ -76,21 +75,30 @@ func (f *sourceCategoryFiller) fill(attributes *pcommon.Map) {
 		return
 	}
 
-	valueTemplate := getAnnotationAttributeValue(f.annotationPrefix, sourceCategorySpecialAnnotation, attributes)
 	var templateAttributes []string
-	if valueTemplate != "" {
-		templateAttributes = extractTemplateAttributes(valueTemplate)
-	} else {
+	doesUseAnnotation := false
+
+	valueTemplate := getAnnotationAttributeValue(f.annotationPrefix, sourceCategorySpecialAnnotation, attributes)
+	if valueTemplate == "" {
 		valueTemplate = f.valueTemplate
-		templateAttributes = f.templateAttributes
+	} else {
+		doesUseAnnotation = true
 	}
 
 	prefix := getAnnotationAttributeValue(f.annotationPrefix, sourceCategoryPrefixAnnotation, attributes)
 	if prefix == "" {
 		prefix = f.prefix
+	} else {
+		doesUseAnnotation = true
 	}
 	valueTemplate = prefix + valueTemplate
 
+	if doesUseAnnotation {
+		templateAttributes = extractTemplateAttributes(valueTemplate)
+	} else {
+		templateAttributes = f.templateAttributes
+	}
```

Comment:
Is it expected that we skip prefix here?

Reply:
`valueTemplate` already contains the prefix, so all templates in the prefix will be affected.

Reply:
I was asking about `f.templateAttributes`. Does it contain correct prefix as well?

Reply:
Yes, see line `41`